modules/dxp/apps/scim/scim-client/src/main/java/com/liferay/scim/client/internal/spi/bearer/token/provider/SCIMClientBearerTokenProvider.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.scim.client.internal.spi.bearer.token.provider;
+
+import com.liferay.oauth2.provider.rest.spi.bearer.token.provider.BearerTokenProvider;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.io.BigEndianCodec;
+import com.liferay.portal.kernel.security.SecureRandomUtil;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Olivér Kecskeméty
+ */
+@Component(service = BearerTokenProvider.class)
```

Comment:
This cannot be declared as a DS @Compoent because then it will override DefaultBearerTokenProvider in all cases